lib/Languages/Galach/Generators/Native/Range.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace QueryTranslator\Languages\Galach\Generators\Native;
+
+use LogicException;
+use QueryTranslator\Languages\Galach\Generators\Common\Visitor;
+use QueryTranslator\Languages\Galach\Values\Node\Term;
+use QueryTranslator\Languages\Galach\Values\Token\Range as RangeToken;
+use QueryTranslator\Values\Node;
+
+/**
+ * Range Node Visitor implementation.
+ */
+final class Range extends Visitor
+{
+    public function accept(Node $node)
+    {
+        return $node instanceof Term && $node->token instanceof RangeToken;
+    }
+
+    public function visit(Node $node, Visitor $subVisitor = null, $options = null)
+    {
+        if (!$node instanceof Term) {
+            throw new LogicException(
+                'Implementation accepts instance of Term Node'
+            );
+        }
+
+        $token = $node->token;
+
+        if (!$token instanceof RangeToken) {
+            throw new LogicException(
+                'Implementation accepts instance of Range Token'
+            );
+        }
+
+        $domainPrefix = '' === $token->domain ? '' : "{$token->domain}:";
+
+        switch ($token->type) {
+            case RangeToken::TYPE_INCLUSIVE:
+                return $domainPrefix . '[' . $token->rangeFrom . ' TO ' . $token->rangeTo . ']';
```

Comment:
Symbols should be captured by the expression and contained in the token, so if they are customized this class does not need to know about it. And we should support mixed case as well, `{a TO b]` and `[a TO b}`.

It will probably mean a truckload of constructor arguments, but I'm OK with that :)